libgalois/src/PerThreadStorage.cpp

```diff
@@ -37,17 +37,27 @@ galois::substrate::PerBackend& galois::substrate::getPPSBackend() {
   return b;
 }
 
-#define MORE_MEM_HACK
 #ifdef MORE_MEM_HACK
-const size_t allocSize =
+const size_t ptAllocSize =
     16 * (2 << 20); // galois::runtime::MM::hugePageSize * 16;
-inline void* alloc() { return malloc(allocSize); }
-
+inline void* alloc() {
+  void* toReturn = malloc(ptAllocSize);
+  if (toReturn == nullptr) {
+    GALOIS_DIE("Out of memory in per thread storage allocation");
+  }
+  return toReturn;
+}
 #else
-const size_t allocSize = galois::runtime::MM::hugePageSize;
-inline void* alloc() { return galois::substrate::MM::pageAlloc(); }
+const size_t ptAllocSize = galois::substrate::allocSize();
+inline void* alloc() {
+  // alloc a single page, don't prefault
+  void* toReturn = galois::substrate::allocPages(1, false);
```

Comment:
why not prefault? seems better to me to fault in now during initialization rather than possibly contending for the vm lock during parallel execution.